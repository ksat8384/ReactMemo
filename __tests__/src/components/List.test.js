import React from "react";
import { Text } from "react-native";
import { render, cleanup } from "@testing-library/react-native";
import List from "../../../src/components/List";

describe("List Component", () => {
  let component;
  const mockList = [
    { id: "a", name: "Batman" },
    { id: "b", name: "Superman" },
    { id: "c", name: "Ironman" },
    { id: "d", name: "Hulk" },
    { id: "e", name: "BlackWidow" },
    { id: "f", name: "Thor" },
    { id: "g", name: "Captain America" },
    { id: "h", name: "Hawkeye" },
    { id: "i", name: "Spiderman" },
    { id: "j", name: "Antman" },
  ];

  beforeEach(() => {
    component = render(<List list={mockList} />);
  });

  afterEach(cleanup);

  it("renders list correctly", () => {
    const { findByTestId, toJSON } = component;
    expect(toJSON()).toMatchSnapshot();
    expect(findByTestId("flat-list")).toBeTruthy();
  });
});
