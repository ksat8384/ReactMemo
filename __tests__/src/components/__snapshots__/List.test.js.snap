// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders list correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "id": "a",
        "name": "Batman",
      },
      Object {
        "id": "b",
        "name": "Superman",
      },
      Object {
        "id": "c",
        "name": "Ironman",
      },
      Object {
        "id": "d",
        "name": "Hulk",
      },
      Object {
        "id": "e",
        "name": "BlackWidow",
      },
      Object {
        "id": "f",
        "name": "Thor",
      },
      Object {
        "id": "g",
        "name": "Captain America",
      },
      Object {
        "id": "h",
        "name": "Hawkeye",
      },
      Object {
        "id": "i",
        "name": "Spiderman",
      },
      Object {
        "id": "j",
        "name": "Antman",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  testID="flat-list"
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Batman
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Superman
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Ironman
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Hulk
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          BlackWidow
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Thor
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Captain America
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Hawkeye
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Spiderman
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#00ffff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Antman
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
